src/mmtk-gc.c

```diff
@@ -53,6 +53,17 @@ static inline void malloc_maybe_collect(jl_ptls_t ptls, size_t sz)
     }
 }
 
+// allocation
+int jl_gc_classify_pools(size_t sz, int *osize)
+{
+    if (sz > GC_MAX_SZCLASS)
+        return -1;
+    size_t allocsz = sz + sizeof(jl_taggedvalue_t);
+    int klass = jl_gc_szclass(allocsz);
+    *osize = LLT_ALIGN(allocsz, 16);
+    return (int)(intptr_t)(&((jl_ptls_t)0)->heap.norm_pools[klass]);
```

Comment:
Consider either removing `jl_gc_classify_pools` (and modifying the code in `llvm-final-gc-lowering.cpp`), or modifying the code in this method so it does not rely on the Julia size class. Maybe you also want to change the check `if (sz > GC_MAX_SZCLASS)` to use a large object threshold from MMTk (I don't know if Julia would work fine with it).

```suggestion
    return 0; // return a non-negative value, to indicate
```